examples/sources/CustomSource.md

```diff
@@ -0,0 +1,24 @@
+# How to add a spout as a custom source
+
+If you already have a spout implemented and want to use it in a SAM application, SAM lets you register it once  and re-use in any SAM application. This 
+way, you can avoid rewriting code and leverage existing spout code. Provided below, is an illustration of how to build and register such a spout as a SAM source. 
+All the files needed to build, register and run are present in this module.
+
+1. Check out the module on your local directory and run `mvn clean install`
+2. Open the `file-source-topology-component.json` file under resources directory and make sure that the mavenDeps property(which is of the form 
+groupId:artifactId:version) has the correct value for version. Version should be the version defined in the pom file of this module
+3. Export the variable `data`. Its value should be the location of the json file edited in step 2 above.
+4. Run the below curl command to register it as a SAM source. Before doing that, make sure you replace sam_host and sam_port with values for your environment
+ and that bundleJar form parameter points to the jar file that you build in step 1. 
+
+`curl -sS -X POST -i -F topologyComponentBundle=@$data  -F bundleJar=@/tmp/custom_source_example/streamline-examples-sources-0.1.0-SNAPSHOT.jar http://<sam_host>:<sam_port>/api/v1/catalog/streams/componentbundles/SOURCE`
+
+5. Install the jar built in step 1 above in local m2 directory for the user running SAM server on SAM server using the command below. Make sure the version 
```

Comment:
Is this required? Doesn't streamline automatically ship the jar (streamline-examples-sources.jar) with the storm jar command (--artifacts) ? Its not very practical for users to install the jar on all the servers running SAM.

Reply:
This is not required for this example. Flux transformation is expected to be in streamline-examples-sources.jar. However, the actual storm spout or bolt does not have to be in the same jar since we are leveraging the mavenDeps string to fetch it from maven. I wanted to highlight that as well in this example.

Reply:
@priyank5485 , so its not required for this example since you are already putting FileReaderSpout as part of the module correct? If so we can remove the step 5 from the README. 

Maybe as a note somewhere at the end you can specify that if storm spout is not included as part of the module you need to install that separately. In this case can they build a fat jar by just having a dependency on their spout in the pom ? It will easier than installing the spout jar in all the server hosts.

Reply:
@arunmahadevan I have pushed a change to readme file as per your suggestion. I did not get your question about fat jar. If the spout is in a separate jar then all they need to make sure is that its either available in maven local or at remote maven repo using the groupId, artifactId and version that they set in mavenDeps. We leverage that property of a non built-in bundle while submitting the topology to storm using --artifact argument to storm.py. That way storm will make sure that it will fetch the jar. Currently, the remote repos used are provided by mavenRepoUrl in streamline.yaml configurable via Ambari. By default maven central is always checked for. They can either have it published to maven central or change mavenRepoUrl and restart SAM or install it in maven local. Hope this answers your question.